File: AccountService/src/main/java/com/app/AccountService/DTO/Response/UserResponse.java
package com.app.AccountService.DTO.Response;

import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class UserResponse {
    String userID;
    String name;
    String phone;
    String gmail;
    List<String> roles;
}

File: ReactTS/food-app-v2/src/APIs/dto/responses/UserResponse.ts
export interface Role {
  name: string;
  description: string;
  permissions: Permission[];
}
export interface Permission {
  name: string;
  description: string;
}

export interface UserResponse {
  userID: string;
  name: string;
  password: string;
  phone: string;
  gmail: string;
  roles: Role[];
}
